src/Workspaces/Core/Portable/FindSymbols/FindReferences/FindReferenceCache.cs

```diff
@@ -32,29 +32,44 @@ public static async ValueTask<FindReferenceCache> GetCacheAsync(Document documen
 
         static async Task<FindReferenceCache> ComputeCacheAsync(Document document, CancellationToken cancellationToken)
         {
+            var text = await document.GetValueTextAsync(cancellationToken).ConfigureAwait(false);
+
             // Find-Refs is not impacted by nullable types at all.  So get a nullable-disabled semantic model to avoid
             // unnecessary costs while binding.
             var model = await document.GetRequiredNullableDisabledSemanticModelAsync(cancellationToken).ConfigureAwait(false);
             var root = await document.GetRequiredSyntaxRootAsync(cancellationToken).ConfigureAwait(false);
-            return new(document, model, root);
+
+            // It's very costly to walk an entire tree.  So if the tree is simple and doesn't contain
+            // any unicode escapes in it, then we do simple string matching to find the tokens.
+            var index = await SyntaxTreeIndex.GetRequiredIndexAsync(document, cancellationToken).ConfigureAwait(false);
```

Comment:
similarly, we will have practically always computed thsi for all documents that were hits anyways (because that's how we figured out there might be a hit in the first place).  So deferring this doesn't help at all, and this elides async from everything that follows.